src/safeposix/syscalls/fs_calls.rs

```diff
@@ -26,158 +59,225 @@ impl Cage {
+                                        occ.remove_entry();
+                                    }
 
-                if let interface::RustHashEntry::Vacant(vac) = FILEOBJECTTABLE.entry(newinodenum) {
-                    let sysfilename = format!("{}{}", FILEDATAPREFIX, newinodenum);
-                    vac.insert(interface::openfile(sysfilename, 0).unwrap()); // new file of size 0
-                }
+                                    // The current file is removed from the filesystem
+                                    let sysfilename = format!("{}{}", FILEDATAPREFIX, inodenum);
+                                    interface::removefile(sysfilename.clone()).unwrap();
+                                }
 
-                let _insertval =
-                    fdoption.insert(File(self._file_initializer(newinodenum, flags, 0)));
-            }
+                                // Once the metadata for the file is reset, a new file is inserted in file system.
+                                // Also, it is inserted back to the FileObjectTable and associated with same inodeNumber representing that the file is currently in open state.
+                                if let interface::RustHashEntry::Vacant(vac) = FILEOBJECTTABLE.entry(inodenum)
+                                {
+                                    let sysfilename = format!("{}{}", FILEDATAPREFIX, inodenum);
+                                    vac.insert(interface::openfile(sysfilename, f.size).unwrap());
+                                }
 
-            //If the file exists (we don't need to look at parent here)
-            (Some(inodenum), ..) => {
-                if (O_CREAT | O_EXCL) == (flags & (O_CREAT | O_EXCL)) {
-                    return syscall_error(
-                        Errno::EEXIST,
-                        "open",
-                        "file already exists and O_CREAT and O_EXCL were used",
-                    );
-                }
-                let size;
+                                // Update the final size and reference count for the file 
+                                size = f.size;
+                                f.refcount += 1;
 
-                let mut inodeobj = FS_METADATA.inodetable.get_mut(&inodenum).unwrap();
-                match *inodeobj {
-                    Inode::File(ref mut f) => {
-                        if O_TRUNC == (flags & O_TRUNC) {
-                            // We only do this to regular files, otherwise O_TRUNC is undefined
-                            //close the file object if another cage has it open
-                            let entry = FILEOBJECTTABLE.entry(inodenum);
-                            if let interface::RustHashEntry::Occupied(occ) = &entry {
-                                occ.get().close().unwrap();
+                                // Doubt: Why are we removing the previous entry from the FileObjectTable and then inserting a new file with size 0. 
```

Comment:
This is  hacky way to truncate since a new file will be size 0 instead of actually adjusting the filesize and pointer. I would comment that thats what were doing but also label it as a possible bug.

Reply:
Updated the comment.